contracts/DelFiPrice.sol

```diff
@@ -27,17 +32,58 @@ contract DelFiPrice is OpenOracleView {
     /// @notice The mapping of medianized prices per symbol
     mapping(string => uint64) public prices;
 
+    /// @notice Address of the cEther contract
+    address public cEthAddress;
+
+    /// @notice Address of the cUSDC contract
+    address public cUsdcAddress;
+
+    /// @notice Address of the cDAI contract
+    address public cDaiAddress;
+
+    /// @notice Address of the cREP contract
+    address public cRepAddress;
+
+    /// @notice Address of the cWBTC contract
+    address public cWbtcAddress;
+
+    /// @notice Address of the cBAT contract
+    address public cBatAddress;
+
+    /// @notice Address of the cZRX contract
+    address public cZrxAddress;
+
     /**
      * @param data_ Address of the Oracle Data contract
      * @param sources_ The reporter addresses whose prices will be used to calculate the median
      * @param anchor_ The reporter address whose prices checked against the median for safety
      * @param anchorToleranceMantissa_ The tolerance allowed between the anchor and median. A tolerance of 10e16 means a new median that is 10% off from the anchor will still be saved
+     * @param cEthAddress_ The address of cETH token asset
+     * @param cDaiAddress_ The address of cDAI token asset
+     * @param cZrxAddress_ The address of cZRX token asset
      */
-    constructor(OpenOraclePriceData data_, address[] memory sources_, address anchor_, uint anchorToleranceMantissa_) public OpenOracleView(data_, sources_) {
+    constructor(OpenOraclePriceData data_, 
+                address[] memory sources_,
+                address anchor_,
+                uint anchorToleranceMantissa_,
+                address cEthAddress_,
+                address cUsdcAddress_,
+                address cDaiAddress_,
+                // address cRepAddress_,
+                // address cWbtcAddress_, 
+                // address cBatAddress_, 
+                address cZrxAddress_) public OpenOracleView(data_, sources_) {
```

Comment:
We can use a list if we need to (like sources), its just hard to do ordering - slightly surprised this doesn't work though since I think we have other constructors with as many args as this

Reply:
3 parameters too many :(  

```
Got this Error: Command failed: solc --combined-json bin,abi,metadata --metadata-literal --optimize  contracts/*.sol
Exception during compilation: /solidity/libyul/backends/evm/AsmCodeGen.cpp(211): Throw in function static void solidity::yul::CodeGenerator::assemble(const solidity::yul::Block&, solidity::yul::AsmAnalysisInfo&, solidity::evmasm::Assembly&, solidity::langutil::EVMVersion, const solidity::yul::ExternalIdentifierAccess&, bool, bool)
Dynamic exception type: boost::wrapexcept<solidity::yul::YulAssertion>
std::exception::what: Stack too deep when compiling inline assembly: Variable headStart is 2 slot(s) too deep inside the stack.
[solidity::util::tag_comment*] = Stack too deep when compiling inline assembly: Variable headStart is 2 slot(s) too deep inside the stack.
```

Reply:
Tried to compile with --optimize-yul flag, but no luck, the same error

Reply:
struct arg could come out nice, or a 2d array [[ctokens], [symbols]] or [[ctoken, symbol], [ctoken, symbol]] are a couple of other ways we may be able to slip out of this one

Reply:
makes sense, I will try both approaches.

Reply:
I added a second PR that gets an array of pairs in constructor. The final result is in a way much simpler than this original Pr.

Reply:
Moving one with this PR and immutable constants.